contracts/extensions/FundingQueue.sol

```diff
@@ -0,0 +1,391 @@
+
+    require(proposal.state != ProposalState.Cancelled, "funding-queue-already-cancelled");
+    require(proposal.state != ProposalState.Completed, "funding-queue-already-completed");
+    require(proposal.creator == msg.sender, "funding-queue-not-creator");
+    require(queue[_prevId] == _id, "funding-queue-bad-prev-id");
+
+    proposal.state = ProposalState.Cancelled;
+    proposal.lastUpdated = now;
+
+    queue[_prevId] = queue[_id];
+    delete queue[_id];
+
+    proposals[queue[_prevId]].lastUpdated = now;
+
+    emit ProposalCancelled(_id);
+  }
+
+  function stakeProposal(
+    uint256 _id,
+    bytes memory _key,
+    bytes memory _value,
+    uint256 _branchMask,
+    bytes32[] memory _siblings
+  )
+    public
+  {
+    Proposal storage proposal = proposals[_id];
+
+    require(proposal.state == ProposalState.Inactive, "funding-queue-not-inactive");
+    require(proposal.creator == msg.sender, "funding-queue-not-creator");
+
+    proposal.state = ProposalState.Active;
+    proposal.domainTotalRep = checkReputation(_id, address(0x0), _key, _value, _branchMask, _siblings);
+
+    uint256 stake = wmul(proposal.domainTotalRep, STAKE_FRACTION);
+    colony.obligateStake(msg.sender, proposal.domainId, stake);
+
+    emit ProposalStaked(_id, proposal.domainTotalRep);
+  }
+
+  function backProposal(
+    uint256 _id,
+    uint256 _backing,
+    uint256 _currPrevId,
+    uint256 _newPrevId,
+    bytes memory _key,
+    bytes memory _value,
+    uint256 _branchMask,
+    bytes32[] memory _siblings
+  )
+    public
+  {
+    Proposal storage proposal = proposals[_id];
+
+    require(proposal.state == ProposalState.Active, "funding-queue-proposal-not-active");
+    require(_id != _newPrevId, "funding-queue-cannot-insert-after-self"); // NOTE: this may be redundant
+
+    uint256 userRep = checkReputation(_id, msg.sender, _key, _value, _branchMask, _siblings);
+    require(_backing <= userRep, "funding-queue-insufficient-reputation");
+
+    // Update the user's reputation backing
+    uint256 prevBacking = supporters[_id][msg.sender];
+    if (_backing >= prevBacking) {
+      proposal.totalSupport = add(proposal.totalSupport, sub(_backing, prevBacking));
+    } else {
+      proposal.totalSupport = sub(proposal.totalSupport, sub(prevBacking, _backing));
+    }
+    supporters[_id][msg.sender] = _backing;
+
+    // Remove the proposal from its current position, if exists
+    require(queue[_currPrevId] == _id, "funding-queue-bad-prev-id");
+    queue[_currPrevId] = queue[_id];
+
+    // Insert into the right location
+    uint256 nextId = queue[_newPrevId];
+    if (queue[HEAD] == nextId && nextId != 0) {
+      pingProposal(nextId);
+    }
+    // Does this proposal have less than or equal support to the previous proposal?
+    require(
+      proposals[_newPrevId].totalSupport >= proposal.totalSupport,
+      "funding-queue-excess-support"
+    );
+    // Does this proposal have more than or equal support to the next proposal?
```

Comment:
Only more than, not equal. The code is right, and having the inequalities this way around between the two requires is good as it means that proposals with the same backing can't jostle for position (effectively, the first proposal to reach a particular amount of backing has priority).

Reply:
Right right, I went back and forth on this a bit yesterday and decided the way I had it was best. I changed the comments and seems I forgot to change them back.